src/resolve/resolved-dns-stub.c

```diff
@@ -727,8 +727,8 @@ static int dns_stub_patch_bypass_reply_packet(
         }
 
         /* Ensure we don't pass along an untrusted ad flag for bypass packets */
-        if (!authenticated)
-                DNS_PACKET_HEADER(c)->flags = htobe16(be16toh(DNS_PACKET_HEADER(c)->flags) & ~DNS_PACKET_FLAG_AD);
+        DNS_PACKET_HEADER(c)->flags = htobe16(be16toh((DNS_PACKET_HEADER(c)->flags) & ~DNS_PACKET_FLAG_AD) |
+                        (authenticated ? DNS_PACKET_FLAG_AD : 0));
```

Comment:
Uhh, not a big matter, but also line break is spurious. How about
```
DNS_PACKET_HEADER(c)->flags = htobe16(
                be16toh((DNS_PACKET_HEADER(c)->flags) & ~DNS_PACKET_FLAG_AD) |
                (authenticated ? DNS_PACKET_FLAG_AD : 0));
```
or
```
DNS_PACKET_HEADER(c)->flags = htobe16(be16toh((DNS_PACKET_HEADER(c)->flags) & ~DNS_PACKET_FLAG_AD) |
                                      (authenticated ? DNS_PACKET_FLAG_AD : 0));
```
?

Reply:
ok #1 looks good